.github/workflows/tests.yaml

```diff
@@ -0,0 +1,127 @@
+name: tests
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["main"]
+
+jobs:
+  unit-test:
+
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # ratchet:actions/checkout@v3
```

Comment:
Side note, what is ratchet?

I know dependabot and renovate

Reply:
https://github.com/sethvargo/ratchet
It's a tool to pin the GitHub actions to specific commit

Reply:
Thanks